akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -936,71 +936,63 @@ private[stream] object Collect {
       private val buffer: BufferImpl[T] = BufferImpl(size, inheritedAttributes)
 
       private val name = inheritedAttributes.nameOrDefault(getClass.toString)
-      val enqueueAction: T => Unit =
+
+      private def enqueueAction(elem: T): Unit =
```

Comment:
This instead adds the overhead of doing the strategy switch for each element, which was previously done up front. Not sure the memory tradeoff is better.

Reply:
Yes, that true.